### PR TITLE
[After #196][service] Add document for newly added service APIs @open sesame 08/31 11:35

### PR DESCRIPTION
**Pull request by anyj0527**
- Add some document for newly added service APIs.
- Change API names
  `ml_service_getstate_pipeline` -> `ml_service_get_pipeline_state`
  `ml_service_destroy_pipeline` -> `ml_service_destroy`
- Fix return values and free proxy properly.

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Yongjoo Ahn <yongjoo1.ahn@samsung.com>

### CODE REVIEW

File: c/include/ml-api-service.h [hunk 1]
```diff
@@ -117,15 +117,88 @@ int ml_service_delete_pipeline (const char *name);
...
+ * @retval #ML_ERROR_INVALID_PARAMETER Fail. The parameter is invalid.
+ * @retval #ML_ERROR_STREAMS_PIPE Failed to access the pipeline state.
+ */
 int ml_service_getstate_pipeline (ml_service_h handle, ml_pipeline_state_e *state);
+
+/**
+ * @brief Gets the description of given handle's pipeline.
+ * @since_tizen 7.0
+ * @param[in] handle The service handle.
+ * @param[out] desc The pipeline description.
```
Comment (myungjoo) [inline c/include/ml-api-service.h:196]: describe whether the caller should free desc or not.

Comment (anyj0527) [reply c/include/ml-api-service.h:198]: Thanks! I added a remarks tag. PTAL

**Review by gichan-jang — APPROVED** [review]
LGTM

**Review by myungjoo — APPROVED** [review]
LGTM

File: c/include/ml-api-service.h [hunk 2]
```diff
@@ -117,16 +117,78 @@ int ml_service_delete_pipeline (const char *name);
...
+ * @since_tizen 7.0
  */
 typedef void *ml_service_h;
 
-/** @todo Add description and decide final API name */
+/**
+ * @brief Launches the pipeline of given service and get the handle.
+ * @details This requests machine learning agent daemon to launch a new pipeline of given service. The pipeline of service @a name should be set.
+ * @since_tizen 7.0
+ * @remarks The @a handle should be destroyed using ml_serice_destroy_pipeline().
```
Comment (wiertel) [inline c/include/ml-api-service.h:128]: typo in function name, or maybe you meant ml_service_destroy() which destroys pipeline?

File: c/include/ml-api-service.h [hunk 3]
```diff
@@ -117,16 +117,78 @@ int ml_service_delete_pipeline (const char *name);
 
 /**
  * @brief handle for ml_service. To support multiple instance for a single service, user should mangage pipeline with this handle.
```
[the remark this answers is not in the document]
Comment (anyj0527) [reply c/include/ml-api-service.h]: Thanks for thorough review!
I changed the sentence with `@brief A handle for ml-service instance.`, removing the ambiguous last sentence.
And for the lack of information, we expect that users could know it from following guide of other APIs. PTAL.

File: c/include/ml-api-service.h (hunk 2 above)
Comment (anyj0527) [reply c/include/ml-api-service.h]: I fixed it to `ml_service_destroy`. Thanks.

File: c/include/ml-api-service.h [hunk 4]
```diff
@@ -117,16 +117,78 @@ int ml_service_delete_pipeline (const char *name);
...
+ * @since_tizen 7.0
  */
 typedef void *ml_service_h;
 
-/** @todo Add description and decide final API name */
+/**
+ * @brief Launches the pipeline of given service and get the handle.
```
[the remark this answers is not in the document]
Comment (anyj0527) [reply c/include/ml-api-service.h]: Fixed, thanks.

File: c/include/ml-api-service.h [hunk 5]
```diff
@@ -117,16 +117,78 @@ int ml_service_delete_pipeline (const char *name);
...
+ * @retval #ML_ERROR_INVALID_PARAMETER Fail. The parameter is invalid.
+ * @retval #ML_ERROR_STREAMS_PIPE Failed to stop the pipeline.
+ */
 int ml_service_stop_pipeline (ml_service_h handle);
-int ml_service_destroy_pipeline (ml_service_h handle);
-int ml_service_getstate_pipeline (ml_service_h handle, ml_pipeline_state_e *state);
-int ml_service_getdesc_pipeline (ml_service_h handle, char **desc);
+
+/**
+ * @brief Destroys the pipeline of given service handle.
```
[the remark this answers is not in the document]
Comment (anyj0527) [reply c/include/ml-api-service.h]: You're right. I changed those as
```
...
 * @brief Destroys the given service handle.
 * @details If given service handle is created by ml_service_launch_pipeline(), this requests machine learning agent daemon to destroy the pipeline.
...
```
PTAL

**Pull request merged**